src/rpm_ostree/cli_status.rs

```diff
@@ -170,41 +159,26 @@ fn status_json(client: &mut RpmOstreeClient) -> Fallible<StatusJSON> {
 }
 
 /// CLI executor for `rpm-ostree status --json`.
-pub fn invoke_cli_status(booted_only: bool) -> Fallible<StatusJSON> {
+pub fn invoke_cli_status() -> Fallible<rpmostree_client::Status> {
     RPM_OSTREE_STATUS_ATTEMPTS.inc();
 
-    let mut cmd = std::process::Command::new("rpm-ostree");
-    cmd.arg("status").env("RPMOSTREE_CLIENT_ID", "zincati");
-
-    // Try to request the minimum scope we need.
-    if booted_only {
-        cmd.arg("--booted");
-    }
-
-    let cmdrun = cmd
-        .arg("--json")
-        .output()
-        .with_context(|_| "failed to run 'rpm-ostree' binary")?;
-
-    if !cmdrun.status.success() {
-        RPM_OSTREE_STATUS_FAILURES.inc();
-        bail!(
-            "rpm-ostree status failed:\n{}",
-            String::from_utf8_lossy(&cmdrun.stderr)
-        );
+    match rpmostree_client::query_status() {
+        Ok(s) => Ok(s),
+        Err(e) => {
+            RPM_OSTREE_STATUS_FAILURES.inc();
+            Err(e).map_err(failure::Error::from_boxed_compat)
+        }
     }
-    let status: StatusJSON = serde_json::from_slice(&cmdrun.stdout)?;
-    Ok(status)
 }
 
 #[cfg(test)]
 mod tests {
     use super::*;
 
-    fn mock_status(path: &str) -> Fallible<StatusJSON> {
+    fn mock_status(path: &str) -> Fallible<rpmostree_client::Status> {
```

Comment:
It's not clear to me we even need these tests anymore, since the upstream library has this.